client/rest-high-level/src/main/java/org/elasticsearch/client/IndicesClient.java

```diff
@@ -786,33 +786,70 @@ public void validateQueryAsync(ValidateQueryRequest validateQueryRequest, Reques
     }
 
     /**
-     * Gets index templates using the Index Templates API
+     * Gets index templates which includes deprecated doc types using the Index Templates API
      * See <a href="https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-templates.html"> Index Templates API
      * on elastic.co</a>
      * @param getIndexTemplatesRequest the request
      * @param options the request options (e.g. headers), use {@link RequestOptions#DEFAULT} if nothing needs to be customized
      * @return the response
      * @throws IOException in case there is a problem sending the request or parsing back the response
+     * @deprecated use {@link #getIndexTemplate(GetIndexTemplatesRequest, RequestOptions)} instead
```

Comment:
Small comment: maybe we could use the same wording as we do in `putMapping` for consistency: https://github.com/elastic/elasticsearch/pull/37280/files#diff-944e3b65663d286958065a2ebc8d1aedR178